core/src/main/java/com/hamdan/forzenbook/core/UtilFunctions.kt

```diff
@@ -46,20 +45,23 @@ fun getBitmapFromUri(
         .into(object : SimpleTarget<Bitmap>() {
             override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
                 // Save the downloaded bitmap to a temporary file
-                val tempFile: File
+                var tempFile: File? = null
                 try {
                     tempFile = createTemporaryImageFile()
                     FileOutputStream(tempFile).use { fos ->
                         resource.compress(Bitmap.CompressFormat.JPEG, 100, fos)
                     }
-                    onBitmapLoaded(resource)
-                    onImageSaved(tempFile)
                 } catch (e: IOException) {
                     e.printStackTrace()
-                    onError()
                 }
+                onImageSaved(tempFile)
             }
         })
 }
 
-private fun createTemporaryImageFile(): File = File.createTempFile(GlobalConstants.TEMPORARY_FILENAME, ".jpg")
+private fun createTemporaryImageFile(): File =
+    File.createTempFile(GlobalConstants.TEMPORARY_FILENAME, ".jpg")
```

Comment:
How do you know it's a jpg?

Reply:
I do not know, I thought since we're retrieveing it as a bitmap to begin with it'd be fine to always have it be a jpg. What is the better approach for something like this? Do I check the ending of the content uri? Though I'm pretty sure that doesn't have the file type at the end of it from when I was messing with it that one day.

Reply:
Not all bitmaps are jpgs. You get the content uri and then you write it to a file, so perhaps during the writing process you can specify it always writes as a certain image format

Reply:
I think that is whats happening right now isn't it? I understand that the file were creating here is assuming jpg but the lines above wit are compressing the bitmap to a jpeg format so it would be valid no?

Reply:
I suppose so